openeo/extra/job_management.py

```diff
@@ -437,15 +470,29 @@ def _update_statuses(self, df: pd.DataFrame):
                 con = self._get_connection(backend_name)
                 the_job = con.job(job_id)
                 job_metadata = the_job.describe()
-                _log.info(
-                    f"Status of job {job_id!r} (on backend {backend_name}) is {job_metadata['status']!r}"
-                )
-                if job_metadata["status"] == "finished":
-                    self.on_job_done(the_job, df.loc[i])
+
+                previous_status = df.loc[i, "status"]  
+                new_status = job_metadata["status"] 
+
+                _log.info(f"Status of job {job_id!r} (on backend {backend_name}) is {new_status!r}")
+
+                if previous_status in {"created", "queued", "started"} and new_status == "running":
+                    df.loc[i, "running_start_time"] = rfc3339.utcnow() 
+
+                if self.cancel_running_job_after and job_metadata["status"] == "running":
+                    self._cancel_prolonged_job(the_job, df.loc[i])
```

Comment:
I would move this canceling after all the other status checks. 

Because now it allows _cancel_prolonged_job to change the status in `df.loc[i]`, and because of `previous_status = df.loc[i, "status"]` above, things might get confusing/inconsistent very easily

Reply:
I propose to place it right above: if new_status == "canceled":

That way the manager reacts properly on the cancelled job within the same 'loop'